tst/EnergyPlus/unit/LowTempRadiantSystem.unit.cc

```diff
@@ -3563,3 +3563,216 @@ TEST_F(LowTempRadiantSystemTest, calculateUFromISOStandardTest)
+    });
+    ASSERT_TRUE(process_idf(idf_objects));
+
+    Zone.allocate(3);
+    Zone(1).Name = "WEST ZONE";
+    Zone(2).Name = "EAST ZONE";
+    Zone(3).Name = "SOUTH ZONE";
+
+    DataSurfaces::TotSurfaces = 6;
+    Surface.allocate(DataSurfaces::TotSurfaces);
+    Surface(1).Name = "ZN001:FLR001";
+    Surface(1).ZoneName = "WEST ZONE";
+    Surface(1).Zone = 1;
+    Surface(1).Construction = 1;
+    Surface(2).Name = "ZN001:FLR002";
+    Surface(2).ZoneName = "WEST ZONE";
+    Surface(2).Zone = 1;
+    Surface(2).Construction = 1;
+    Surface(3).Name = "ZN002:FLR001";
+    Surface(3).ZoneName = "EAST ZONE";
+    Surface(3).Zone = 2;
+    Surface(3).Construction = 1;
+    Surface(4).Name = "ZN002:FLR002";
+    Surface(4).ZoneName = "EAST ZONE";
+    Surface(4).Zone = 2;
+    Surface(4).Construction = 1;
+    Surface(5).Name = "ZN003:FLR001";
+    Surface(5).ZoneName = "SOUTH ZONE";
+    Surface(5).Zone = 3;
+    Surface(5).Construction = 1;
+    Surface(6).Name = "ZN003:FLR002";
+    Surface(6).ZoneName = "SOUTH ZONE";
+    Surface(6).Zone = 3;
+    Surface(6).Construction = 1;
+    state->dataConstruction->Construct.allocate(1);
+    state->dataConstruction->Construct(1).SourceSinkPresent = true;
+
+    GetLowTempRadiantSystem(*state);
+
+    EXPECT_EQ(3, LowTempRadiantSystem::TotalNumOfRadSystems);
+
+    EXPECT_EQ(1, LowTempRadiantSystem::NumOfHydrLowTempRadSys);
+    EXPECT_EQ(1, LowTempRadiantSystem::NumOfHydrLowTempRadSysDes);
+
+    EXPECT_EQ(1, LowTempRadiantSystem::NumOfCFloLowTempRadSys);
+    EXPECT_EQ(1, LowTempRadiantSystem::NumOfCFloLowTempRadSysDes);
+
+    EXPECT_EQ(1, LowTempRadiantSystem::NumOfElecLowTempRadSys);
+
+    EXPECT_EQ("WEST ZONE RADIANT FLOOR", RadSysTypes(1).Name);
+    EXPECT_EQ(LowTempRadiantSystem::SystemType::HydronicSystem, RadSysTypes(1).SystemType);
+    EXPECT_EQ(LowTempRadiantSystem::HydrRadSys(1).ZoneName, "WEST ZONE");
+    EXPECT_EQ(LowTempRadiantSystem::HydrRadSys(1).SurfListName, "WEST ZONE SURFACE GROUP");
+    EXPECT_EQ("WEST ZONE RADIANT FLOOR DESIGN", LowTempRadiantSystem::HydrRadSys(1).designObjectName);
+    EXPECT_EQ(1, LowTempRadiantSystem::HydrRadSys(1).DesignObjectPtr);
+
+    EXPECT_EQ("SOUTH ZONE LOWTEMPRAD", RadSysTypes(2).Name);
+    EXPECT_EQ(LowTempRadiantSystem::SystemType::ConstantFlowSystem, RadSysTypes(2).SystemType);
+    EXPECT_EQ(LowTempRadiantSystem::CFloRadSys(1).ZoneName, "SOUTH ZONE");
+    EXPECT_EQ(LowTempRadiantSystem::CFloRadSys(1).SurfListName, "SOUTH ZONE SURFACE GROUP");
+    EXPECT_EQ("SOUTH ZONE LOWTEMPRAD DESIGN", LowTempRadiantSystem::CFloRadSys(1).designObjectName);
+    EXPECT_EQ(1, LowTempRadiantSystem::CFloRadSys(1).DesignObjectPtr);
+
+    EXPECT_EQ("EAST ZONE RADIANT FLOOR", RadSysTypes(3).Name);
+    EXPECT_EQ(LowTempRadiantSystem::SystemType::ElectricSystem, RadSysTypes(3).SystemType);
+    EXPECT_EQ(LowTempRadiantSystem::ElecRadSys(1).ZoneName, "EAST ZONE");
+    EXPECT_EQ(LowTempRadiantSystem::ElecRadSys(1).SurfListName, "EAST ZONE SURFACE GROUP");
```

Comment:
Check that the assignment goes well. It fails without the fix

Reply:
Fantastic!!